### PR TITLE
Convert cli to esm

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: cli/import-jdl.mjs [hunk 1]
```diff
@@ -16,22 +16,23 @@
...
+import EnvironmentBuilder from './environment-builder.mjs';
+import { CLI_NAME, GENERATOR_NAME, logger, toStringJoinArgs, printSuccess, getOptionAsArgs } from './utils.mjs';
+import { packageJson as packagejs } from '../lib/index.cjs';
+import statistics from '../generators/statistics.cjs';
+import { JHIPSTER_CONFIG_DIR } from '../generators/generator-constants.mjs';
+import { writeConfigFile } from './export-utils.cjs';
+
+const require = createRequire(import.meta.url);
+
+const jhipsterCli = require.resolve('./cli.mjs');
```
Comment (Tcharl) [inline cli/import-jdl.mjs:35]: No solution to use `import()` here?

**Comment by Tcharl** [comment]
Awesome! Thank you so much: the internals of esm are so hard to get

**Comment by mshima** [comment]
Still needs some work.

File: cli/import-jdl.mjs (hunk 1 above)
Comment (mshima) [reply cli/import-jdl.mjs:35]: I will switch to path.join. Import.resolve is experimental.

File: generators/upgrade/generator.mjs [hunk 2]
```diff
@@ -30,10 +30,9 @@ import BaseGenerator from '../base/index.mjs';
 import { upgradeFiles } from '../cleanup.mjs';
 import constants from '../generator-constants.cjs';
 import statistics from '../statistics.cjs';
-import blueprintUtils from '../../utils/blueprint.cjs';
+import { parseBluePrints } from '../../utils/blueprint.mjs';
```
Comment (Tcharl) [inline generators/upgrade/generator.mjs:33]: `parseBlueprint`?

Comment (mshima) [reply generators/upgrade/generator.mjs:33]: The original is parseBluePrints.

Comment (Tcharl) [reply generators/upgrade/generator.mjs:33]: Yes, but we're preparing V8 aren't we?

Comment (mshima) [reply generators/upgrade/generator.mjs:33]: Don’t analyzed if we need to rename. Can rename later if makes sense.

**Comment by mshima** [comment]
Further improvements can be done later.
Should be ready now.

**Review by DanielFran — APPROVED** [review]

**Pull request merged**